### DISCUSSION TITLE
French Trad

### DISCUSSION THREAD

**Discussion by Baduk07** [Nimdy/Dedicated_Valheim_Server_Script#157] [Show and tell]
Hello,

I make a french translation for your script.
And i can maintain it easyli.

[menu_fr.zip](https://github.com/Nimdy/Dedicated_Valheim_Server_Script/files/6115296/menu_fr.zip)

**Comment by Nimdy** [comment]
thats really cool! 

However, I have a new menu.sh  and advancement.sh

some of the areas have been removed and sections have been updated.

Could you review the new files and make the changes?

I will then review the code and make sure it works :)

**Reply by Baduk07** [reply]
I use it since this morning :)

i dl new menu.sh and advancement.sh and i translate in the day

**Comment by Nimdy** [comment]
okay. 

You idea is great.

I will add support for all languages...

If you wait. I will redo a lot of the code and all you have to do is edit the fr.config file.

Give me a day or so... I will have something

**Reply by Baduk07** [reply]
ok, good idea

I'm waiting for your green light

**Reply by Nimdy** [reply]
just so you know I didnt forget.... I am working on it....

https://github.com/Nimdy/Dedicated_Valheim_Server_Script/blob/language

You will see I am rewriting the menu script.... it is taking me FOREVER>>>>>>>AS>DA>SD>ASD>AS>D

I think I will only do the main.sh file at this time because of the amount of time it takes to complete this.  I am also making the lang config template file as well.

Once it is complete, then people can translate from a template very easily.

I will keep you posted.

Plus I need to test still... not sure what is breaking and what isnt atm.... I am just making edits as I go through the file.

**Reply by Baduk07** [reply]
no problem, when you're ready, I'll see your message and get to work :)

**Reply by Nimdy** [reply]
moved to the beta tree... This is 100% ready to go.

The template needs templating... is that word?

But I am sure you will understand what to do.

https://github.com/Nimdy/Dedicated_Valheim_Server_Script/tree/beta/lang

**Reply by Baduk07** [reply]
hello, my translate :)
[FR.zip](https://github.com/Nimdy/Dedicated_Valheim_Server_Script/files/6137655/FR.zip)

**Reply by Nimdy** [reply]
added! Thank you.

Please update menu and test.

You will have to change the EN to FR in the menu.sh or advacemenu.sh second line.

I will test tonight, when I get hoem

**Comment by Baduk07** [comment]
I already test on my serveur. Just an error on github. I send the error in the morning. 7h45 am now for me 😁

**Reply by Nimdy** [reply]
i fixed it for you. just update the menu :)

**Reply by Nimdy** [reply]
Added more... Check the bottom of the EN.conf

I added the Valheim+ stuff. 

Moved everything into the advancemenu.sh today...

**Reply by Baduk07** [reply]
Ok, I saw some mistakes in French on my part that I am correcting at the moment.  around 1pm, I test and send you my correction

**Reply by Baduk07** [reply]
Corrected !
[FR.zip](https://github.com/Nimdy/Dedicated_Valheim_Server_Script/files/6141359/FR.zip)

**Reply by Nimdy** [reply]
nice. I added these as well its in the beta branch
https://github.com/Nimdy/Dedicated_Valheim_Server_Script/blob/beta/lang/EN.conf

I updated the template.conf in the beta too


##############################################################
##################START VALHEIM PLUS SECTION##################
##############################################################
FUNCTION_VALHEIM_PLUS_BUILD_CONFIG_INFO="Deleting old configuration if file exist"
FUNCTION_VALHEIM_PLUS_BUILD_CONFIG_INFO_1="Building systemctl instructions for Valheim"
FUNCTION_VALHEIM_PLUS_BUILD_CONFIG_SET_VANILLA="Setting Server for Valheim Vanilla"
FUNCTION_VALHEIM_PLUS_BUILD_CONFIG_SET_PLUS="Setting Server for Mods using Valheim+"

FUNCTION_VALHEIM_PLUS_INSTALL_CHANGING_DIR="Changing into Valheim Install Directory"
FUNCTION_VALHEIM_PLUS_INSTALL_CHECKING_OLD_INSTALL="Checking for older Valheim+ Package files and removing"
FUNCTION_VALHEIM_PLUS_INSTALL_DOWNLOADING_VALHEIM_PLUS_FROM_REPO=""Downloading Latest Valheim+ UnixServer.zip from Official Github"
FUNCTION_VALHEIM_PLUS_INSTALL_CREATING_VER_STAMP="Stamping Current Verison for historics"
FUNCTION_VALHEIM_PLUS_INSTALL_UNPACKING_FILES="Unpacking zip file"
FUNCTION_VALHEIM_PLUS_INSTALL_REMOVING_OLD_BEPINEX_CONFIG="Removing old bepinex config"
FUNCTION_VALHEIM_PLUS_INSTALL_BUILDING_NEW_BEPINEX_CONFIG="Building Start Configuration File for Modded Server with bepinex filled information"
FUNCTION_VALHEIM_PLUS_INSTALL_SETTING_STEAM_OWNERSHIP="Setting steam ownership to Directories, Folders and Files"
FUNCTION_VALHEIM_PLUS_INSTALL_GET_THEIR_VIKING_ON="Who wants to get their Viking mod on HUH!"
FUNCTION_VALHEIM_PLUS_INSTALL_LETS_GO="Let's GO!!!!"

FUNCTION_VALHEIM_PLUS_ENABLE="Enabling Valheim+"
FUNCTION_VALHEIM_PLUS_RESTARTING="Restarting Valheim Services, so changes can be applied"
FUNCTION_VALHEIM_PLUS_ENABLED_ACTIVE="Valheim+ is now enabled and Active" 

FUNCTION_VALHEIM_PLUS_DISABLE="Valheim+ Disable
FUNCTION_VALHEIM_PLUS_DISABLE_RESTARTING="Restarting Valheim Services, so changes can be applied"
FUNCTION_VALHEIM_PLUS_DISABLE_INFO="Valheim+ is now disabled, Vanilla mode Active"

FUNCTION_VALHEIM_PLUS_UPDATE_INFO="Checking for Valheim+ Update"
FUNCTION_VALHEIM_PLUS_UPDATE_NO_UPDATE_FOUND="No update found"
FUNCTION_VALHEIM_PLUS_UPDATE_UPDATE_FOUND="Update found!"
FUNCTION_VALHEIM_PLUS_UPDATE_CONTINUE="Do you wish to continue?"
FUNCTION_VALHEIM_PLUS_UPDATE_BACKING_UP_VPLUS_CONFIG="Making quick backup of valheim_plus.cfg"
FUNCTION_VALHEIM_PLUS_UPDATE_DOWNLOADING_VPLUS="Grabbing Latest from Valheim Plus and Installing!"
FUNCTION_VALHEIM_PLUS_UPDATE_RESTARTING_SERVICES="Restarting Services to apply updates" 
FUNCTION_VALHEIM_PLUS_UPDATE_CANCELED="Canceled the upgrading of Valheim Plus - because Loki sucks" 

FUNCTION_VALHEIM_PLUS_EDIT_VPLUS_CONFIG_SAVE_RESTART="If you made changes, Valheim Services need to be restarted to take effect"
FUNCTION_VALHEIM_PLUS_EDIT_VPLUS_CONFIG_SAVE_RESTART_1="Do you wish to restart Valheim Services now?"
FUNCTION_VALHEIM_PLUS_EDIT_VPLUS_RESTART_SERVICES="Restarting Valheim Server Services"
FUNCTION_VALHEIM_PLUS_EDIT_VPLUS_CANCEL="Canceled Restarting of Valheim Server Service - because Loki sucks"

FUNCTION_VALHEIM_PLUS_EDIT_BEPINEX_CONFIG_RESTART="If you made changes, Valheim Services need to be restarted to take effect"
FUNCTION_VALHEIM_PLUS_EDIT_BEPINEX_CONFIG_RESTART_1="Do you wish to restart Valheim Services now?"
FUNCTION_VALHEIM_PLUS_EDIT_BEPINEX_RESTART_SERVICE_INFO="Restarting Valheim Server Services"
FUNCTION_VALHEIM_PLUS_EDIT_BEPINEX_CANCEL="Canceled Restarting of Valheim Server Service - because Loki sucks"

FUNCTION_VALHEIM_PLUS_MENU_HEADER="Valheim+ Mod Install"
FUNCTION_VALHEIM_PLUS_MENU_INSTALL="Install Valheim Mods"
FUNCTION_VALHEIM_PLUS_MENU_ADMIN_HEADER="Valheim+ Management"
FUNCTION_VALHEIM_PLUS_MENU_ENABLE="Enable Valheim+ on Server"
FUNCTION_VALHEIM_PLUS_MENU_DISABLE="Disable Valheim+ on Server"
FUNCTION_VALHEIM_PLUS_MENU_START="Start Valheim+ Service"
FUNCTION_VALHEIM_PLUS_MENU_STOP="Stop Valheim+ Service"
FUNCTION_VALHEIM_PLUS_MENU_RESTART="Restart Valheim+ Service"
FUNCTION_VALHEIM_PLUS_MENU_STATUS="Status Valheim+ Service"
FUNCTION_VALHEIM_PLUS_MENU_UPDATE="Update Valheim+ File System"
FUNCTION_VALHEIM_PLUS_MENU_MOD_PLUGIN_HEADER="Mod/Plugin Configuration File Editor"
FUNCTION_VALHEIM_PLUS_MENU_MOD_INFO="Mods require Server and Client File Installation"
FUNCTION_VALHEIM_PLUS_MENU_MOD_INFO_1="Editor uses nano, services require restart if"
FUNCTION_VALHEIM_PLUS_MENU_MOD_INFO_2="any changes are made. You will be prompted"
FUNCTION_VALHEIM_PLUS_MENU_MOD_INFO_3="Common Commands: Save: CTRL+o enter Exit: CTRL+x"
FUNCTION_VALHEIM_PLUS_MENU_MOD_INFO_4="Exit without saving: CTRL+x"
FUNCTION_VALHEIM_PLUS_MENU_VPLUS_CONFIG_EDIT="Valheim+ Mod Option Editor"
FUNCTION_VALHEIM_PLUS_MENU_BEPINEX_CONFIG_EDIT="BepInEx Mod Option Editor"
FUNCTION_VALHEIM_PLUS_MENU_RETURN_MAIN="Go to Main Menu"

**Reply by Baduk07** [reply]
ok, back to work :D

**Reply by Baduk07** [reply]
Can i work directly on beta section for FR.conf ?

**Reply by Nimdy** [reply]
if you click the file from there and edit it.  It will then create a repo in your Github.  Then you can make push to mine. So, yes.

**Reply by Baduk07** [reply]
Haaaaaa ! I try that after my work :)

**Comment by Baduk07** [comment]
I have this message when i execute the script, however, the script work perfectly

fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git


I think that is a problem for updating the script from Github

**Reply by Nimdy** [reply]
not sure... that should work... the menu updater acts strange from time to time... I still cant figure out... I never seem to have problems on my end... wish I could reproduce this... maybe there is a better way to update my menu...

**Reply by Nimdy** [reply]
since the menu system doesnt store any information... you could always just delete the folder and re clone it..

just do the git clone -b beta (git url)

**Comment by Nimdy** [comment]
thank you. Merged with main branch now! :)

**Reply by Baduk07** [reply]
I'm an idiot, I didn't click on merge yesterday at noon when I was done ... xD

**Reply by Nimdy** [reply]
never put yourself down for anything.  Thank you for your help! Much appreciated!

**Reply by Baduk07** [reply]
My first github, I am not yet familiar with the site

**Comment by Baduk07** [comment]
Update for the server and the script make the job ! Good job man, you rock !
